pandas/core/arrays/categorical.py

```diff
@@ -2725,17 +2724,15 @@ def _factorize_from_iterable(values):
         If `values` has a categorical dtype, then `categories` is
         a CategoricalIndex keeping the categories and order of `values`.
     """
-    from pandas.core.indexes.category import CategoricalIndex
-
     if not is_list_like(values):
         raise TypeError("Input must be list-like")
 
-    if is_categorical(values):
-        values = CategoricalIndex(values)
-        # The CategoricalIndex level we want to build has the same categories
+    if is_categorical_dtype(values):
```

Comment:
IIRC @jschendel didn't we just change this to make a perf fix? is this change compat with that

Reply:
I put together this version by looking at the old version and taking out the non-executed paths.  So this shouldn't take a hti

Reply:
@jreback : I think you're referring to #27669, which should not be impacted by this.

Reply:
OT: we *might* be able to remove is_categorical entirely?

Reply:
Yah we should.  Will need to be deprecated since its in the API